### ISSUE TITLE
skip hash check doesn't do simple checks?

### ISSUE THREAD

**Issue by sjpotter** [arvidn/libtorrent#5752]
**Please provide the following information**

libtorrent version (or branch): 1.1.13.0

platform/architecture: linux/x86-64

please describe what symptom you see, what you would expect to see instead and
how to reproduce it.

this is my experience with deluge, but I assume it reflects the underlying libtorrent, but if not, please feel free to close.

In deluge, if I do a skip hash check, it assumes the torrent is complete, even if there are no files there, or they are the wrong size.  to me, skip hash check should simply be skip the time / io intensive data check, not simple checks for file existence / correct size.

to me, it would make sense to say

if a file doesn't exist, all the parts that are part of it, should be considered not there.  If a file exists, but is not the right size, I'd think its best to consider it an error, and put the torrent into an error state (i.e. you are saying skip hash check on files that can't possibly be right, do we really want a user to accidentally destroy them?)

as a corollary to this, this might be more specific to deluge, but who knows, when one does a skip hash check, and the contents are not there, it will proceed on its merry seeding way until someone tries to download parts from it, when it will then transition  by itself into a downloading/leeching state.  Again, not what I expect a client to ever do.  It should never transitio by itself from seeding -> leeching.   It should just go into an error state.  If a user wants to then put it into a leeching state, that's their decision to make.

**Comment by an0n666** [comment]
This was addressed in 1.2.8 release. So if you update to the latest libtorrent RC_1_2 release you shouldn't face this issue.

**Comment by sjpotter** [comment]
this commit? https://github.com/arvidn/libtorrent/commit/4ebcf8de4b7fbac450e110412392c330468465c5

**Comment by an0n666** [comment]
There were multiple. If you're trying to backport, that may not apply cleanly.

**Comment by sjpotter** [comment]
not trying to backport, just to understand what was done.

**Comment by arvidn** [comment]
One problem is that 1.1.13 is so old that it would require a fair amount of digging to understand all the changes to the checking logic since then.

The most recent changes to this, in the 1.2 branch was to allow, optionally, to skip checking whether files exists. This is to support the case where the disk a torrent is stored on was unmounted.

**Comment by sjpotter** [comment]
ok, I guess I was trying to figure out what the defaults for skip hash checking are now.  i.e. deluge does this if one does it (not 100% sure what it means)

add_torrent_params['flags'] |= lt.add_torrent_params_flags_t.flag_seed_mode

and that causes the issues I had, perhaps conflating skip hash checking with seed mode is a problem for it?

**Comment by sjpotter** [comment]
as an aside, a related issue to this skip hash, on this version of deluge and libtorrent, if I'm seeding a torrent that I created that I skiped hash check when I loaded it, and lets say I accidentally move it away (actually just happened, meant to move other things, not it).  It will go from seeding to leeching by itself.  I think this behavior is very problematic.  It should error out, not go into leech mode.

has this also been fixed?
